src/app/shared/services/dynamic-data/adapters/persistedMemory.ts

```diff
@@ -48,17 +48,22 @@ const SCHEMA: RxJsonSchema<any> = {
     flow_name: { type: "string", maxLength: 64 },
     flow_type: { type: "string", maxLength: 64 },
     row_id: { type: "string", maxLength: 64 },
+    row_index: { type: "integer", minimum: 0, maximum: 10000, multipleOf: 1, final: true },
     data: {
       type: "object",
     },
   },
-  required: ["id", "flow_type", "flow_name", "row_id", "data"],
-  indexes: ["flow_type", "flow_name", "row_id"],
+  required: ["id", "flow_type", "flow_name", "row_id", "data", "row_index"],
+  indexes: ["flow_type", "flow_name", "row_id", "row_index"],
 };
 const MIGRATIONS: MigrationStrategies = {
   // As part of RXDb v14 update all data requires migrating to change metadata fields (no doc data changes)
   // https://rxdb.info/releases/14.0.0.html
   1: (doc) => doc,
+  2: (oldDoc) => {
+    const newDoc = { ...oldDoc, row_index: 0 };
```

Comment:
+1 - I took a quick look to see if there was any way to get the actual index but not passed to rxdb migration executor so I think your workaround to just make identical and use secondary sort index based on id is the best option